vm/src/builtins.rs

```diff
@@ -312,19 +313,42 @@ fn builtin_id(vm: &VirtualMachine, args: PyFuncArgs) -> PyResult {
 
 // builtin_input
 
-fn builtin_isinstance(obj: PyObjectRef, typ: PyClassRef, vm: &VirtualMachine) -> PyResult<bool> {
-    vm.isinstance(&obj, &typ)
+fn builtin_isinstance(
+    obj: PyObjectRef,
+    typ: Either<PyClassRef, PyTupleRef>,
```

Comment:
Hmm, I'm not sure about the `Either` type. Why not use `PyObjectRef` and use something like this:
```rust
if let cls = to_PyClassRef(typ) {
   ...
} else if let tuple = to_tuple_ref(typ) {
   ...
} else {
   return typeerror('expected class or tuple of classes')
}
```
My problem is with the mapping of the either type into members A and B. This introduces a mapping.

Reply:
I changed it to use the `match_class` macro, which is equivalent to the above.